src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -359,20 +370,32 @@ pub async fn compact(
         .iter()
         .map(|table_info| table_info.file_size)
         .sum::<u64>();
-    match generate_splits(&sstable_infos, compaction_size, context.clone()).await {
-        Ok(splits) => {
-            if !splits.is_empty() {
-                compact_task.splits = splits;
+    let all_ssts_are_blocked_filter = sstable_infos
+        .iter()
+        .all(|table_info| table_info.bloom_filter_kind() == BloomFilterType::Blocked);
+    let optimize_by_copy_block = context.storage_opts.enable_fast_compaction
+        && all_ssts_are_blocked_filter
+        && !has_tombstone
+        && !has_ttl
+        && single_table
+        && compact_task.target_level > 0
+        && compact_task.input_ssts.len() == 2
```

Comment:
Because it is difficult to implement this algorithm of more than 2 sub iterator.
You can see that I only use `left` and `right` two iterator  in `fast_compact_runner::CompactorRunner`.